R/st_bonds.R

```diff
@@ -0,0 +1,192 @@
+#' Sensitivity analysis: `run_stress_test()` multiple times
+#'
+#' These functions performs the so called "sensitivity analysis" for the asset
+#' type in their name. They run [run_stress_test()] multiple times, iterating
+#' over multiple values of one and only one argument of [run_stress_test()].
+#'
+#' @param data Character vector of length 2. Paths to the directories
+#'   ST_INPUTS_MASTER and ST_TESTING_\{aset-type\}, e.g. ST_TESTING_BONDS,
+#'   respectively. These directories must have the files expected by the
+#'   stress-test project. The easiest way to create this object is with
+#'   [st_data_paths()].
+#' @param destdir Character. Path to a destination directory.
+#' @param ... Arguments passed to [run_stress_test()].
+#' @param quiet Logical. Print non-condition messages to the console?
+#'
+#' @seealso [st_data_paths()].
+#' @family main functions
+#'
+#' @return A data frame.
+#'
+#' @export
+#'
+#' @examplesIf r2dii.climate.stress.test:::is_registered_dev("/home/mauro")
+#' library(readr, warn.conflicts = FALSE)
+#'
+#' # Give the paths to your data explicitely
+#' data <- st_data_paths(
+#'   data = "/home/mauro/tmp/st/ST_INPUTS_MASTER",
+#'   project = "/home/mauro/tmp/st/ST_TESTING_BONDS"
+#' )
+#' out <- suppressWarnings(
```

Comment:
I assume out is all data combined in one df? I wonder if this output is not too overwhelming to users to be of use, or if it would, worst case, convey misleading information, as the data result sets are not meant to be considered together.

Reply:
what would be a better option though? if the user passes 10 variations of a parameter, which I consider entirely plausible for a sensitivity analysis, and we save the results to separate files, the output directory will be cluttered and hard to navigate. What is more, the intuitive way to interpret results from a sensitivity analysis would be to visualise the variations in one graph, imo. One df seems like a good solution for that?

Reply:
The concrete problem here is that the out dataframe combines the 5 results types as far as i can see, e.g. market risk and credit risk result.

Reply:
ah okay, if that is the case, that would indeed be hard to digest